js/src/Ice/ObjectPrx.js

```diff
@@ -3,8 +3,8 @@
 //
 
 export class ObjectPrx {
-    constructor() {
-        this._reference = null;
-        this._requestHandler = null;
+    constructor(communicatorOrReference, proxyString = undefined) {
```

Comment:
Can we / should we define two constructors, one with a single param and the other with two?

Reply:
We cannot there is not overloading in JavaScript